src/logger/SanitizeMessage.ts

```diff
@@ -0,0 +1,16 @@
+export const VARIABLES_IN_MESSAGE_REGEX = new RegExp(/got invalid value (.*); Field/gm)
+
+/**
+ * Removes sensible information that might occur when
+ * variables are used in log messages from the message.
+ * @param {string} logMessage - The original log message
+ * @returns {string} The sanitized message. Sensible parts will
+ * be overwritten with the text REMOVED BY SANITIZER
+ */
+export function sanitizeMessage(logMessage: string): string {
+    let foundVariable
+    if (logMessage && (foundVariable = VARIABLES_IN_MESSAGE_REGEX.exec(logMessage))) {
+        return logMessage.replace(foundVariable[1], 'REMOVED BY SANITIZER')
+    }
+    return logMessage
+}
```

Comment:
`RegExp.exec` is a kind of weird function. If you use it with groups (i.e. in our case brackets around ".*") it will store the whole found match in `foundVariable[0]`, the found group in `foundVariable[1]`. As we only want to replace the group (i.e. in our case the full variables output) we use `foundVariable[1]`. The last part after the replaces variables output might contain information about which field caused the issue, if we replace the whole message there is basically no info left what has caused the problem.

Tests for this are located in `CreateLogEntry.test.ts` so we can test that the whole expected log entry is correct.

I'll check the points mentioned above and will add some comment to the function as understanding exec without the knowledge above is quite difficult.

Reply:
Thanks for the explanation 👍